### ISSUE TITLE
MajorBlocksMinor

### ISSUE THREAD

**Issue by adaneslavo** [LoneGazebo/Community-Patch-DLL#8785]
This a feature request, but really serious. The problem is, that each new civilization adapted to the VP and/or MUCfVP must be checked if it doesn't conflict with current City-States f.e. Canada civilization by @pineappledan has two such conflicts: Vancouver and Quebec City. My question is: can we add a table to VP named f.e. MajorBlockMinors, where you define two columns MajorPlayer and Minor Player and it would like this:
```
CIVILIZATION_CANADA          MINOR_CIV_VANCOUVER
CIVILIZATION_CANADA          MINOR_CIV_QUEBEC_CITY
```
Then at some point of map generation and randomizing city-states from the pool, this table would be taken into account and blocked those city states from spawning if CANADA appeared in game. If not, they could spawn without any obstacles.

Such a feature is used in Civ 6 according to @pineappledan. It would make our lives much easier. There are many mods dependent on city-states, and each new civilization created by players (currently there are 6-7 top designers) causes problems and makes other modders program some fixes, patches etc.

Such a feature could be used even for base civs who have their cities farther on the list (Assyria, Babylon).

This would also allow us to add more city-states to the UCS.

**Comment by psparky** [comment]
Good idea, but better still, do it automatically. When picking a CS, just reject it if its name appears in the city list of any civs in the current game (or maybe just their capitals?).

**Comment by adaneslavo** [comment]
I prefer manually. I have mod that changes all game names to endonyms and xenonyms and all of them are dependent on civilization's language and may differ slightly, even by one letter. Such an automatic function wouldn't work then.

**Comment by LessRekkless** [comment]
If you can help us find out **where** in the code a civilization is chosen for each player, it would help out a lot.

As far as I can tell, civilizations are chosen before any function in CvPregame is run.

**Comment by LoneGazebo** [comment]
@LessRekkless you can trace the functions backwards out of CvPregame, and yes, it happens before anything else is set.

**Comment by adaneslavo** [comment]
It would be awesome. It will be used for many modmods, not only mine, so, I cannot wait for that :)
